controllers/infrastructure/byomachine_controller.go

```diff
@@ -414,6 +443,59 @@ func (r *ByoMachineReconciler) setPausedConditionForByoHost(ctx context.Context,
 	return helper.Patch(ctx, machineScope.ByoHost)
 }
 
+func (r *ByoMachineReconciler) setHostInfoForByoMachine(ctx context.Context, machineScope *byoMachineScope) error {
+	helper, err := patch.NewHelper(machineScope.ByoMachine, r.Client)
+	if err != nil {
+		return err
+	}
+
+	machineScope.ByoMachine.Status.HostInfo = machineScope.ByoHost.Status.HostDetails
+
+	return helper.Patch(ctx, machineScope.ByoMachine)
```

Comment:
In Reconciler only, there is a defer function call which will patch ByoMachine. So, patching ByoMachine here is not required.